ichnaea/api/locate/query.py

```diff
@@ -91,6 +92,19 @@ def __init__(
             raise ValueError("Invalid api_type.")
         self.api_type = api_type
 
+        bind_threadlocal(
```

Comment:
`structlog` uses a immutable logger pattern. Binding a variable returns a new logger with the variables attached. In a framework like `Twisted`, you'd pass the logger to view functions and return the new bound logger.

`bind_threadlocal` is part of the [Thread Local Context](https://www.structlog.org/en/stable/thread-local.html) toolkit, which seems like a better fit for our Pyramid views rather than passing a logger around. Binding still replaces the thread local logger, so it makes sense to set as many of these variables at once. For other metrics, I bound them at the same time as the metric was omitted, to make it easier to share logic.